[ML-OB] document python SDK annotation context (#25396)
DataDog/documentation: content/en/llm_observability/setup/sdk/_index.md

{{< /code-block >}}

> ## Modifying auto-instrumented spans

Reviewer: Let's call this `Annotating auto-instrumented spans` to be more clear here

Reviewer: Can we place this as a subheader under `Annotating a span` section?

> The SDK's `LLMObs.annotate_context()` method returns a context manager that can be used to modify auto-instrumented spans started while the annotation context is active.

Reviewer: ```suggestion
The SDK's `LLMObs.annotate_context()` method returns a context manager that can be used to modify all auto-instrumented spans started while the annotation context is active.
```

### Arguments

The `LLMObs.annotation_context()` method accepts the following arguments:

`name` 
: optional - _str_

> <br />Set to override the span name for any auto-instrumented spans started within the annotation context.

Reviewer: ```suggestion
<br />Overrides the span name for any auto-instrumented spans started within the annotation context.
```

`prompt` 
: optional - _dictionary_ 

> <br />A dictionary that represents the prompt used for an LLM call in the following form: `{"template": "...", "id": "...", "version": "...", "variables": {"variable_1": "...", ...}}`. The `Prompt` object can also be imported from `ddtrace.utils` and passed in as the `prompt` argument. **Note**: This argument is only applicable to LLM spans.

Reviewer 2: ```suggestion
<br />A dictionary that represents the prompt used for an LLM call in the following format:<br />`{"template": "...", "id": "...", "version": "...", "variables": {"variable_1": "...", ...}}`.<br />You can also import the `Prompt` object from `ddtrace.utils` and pass it in as the `prompt` argument. **Note**: This argument only applies to LLM spans.
```

`tags`
: optional - _dictionary_

> <br />A dictionary of JSON serializable key-value pairs that users can add as tags on the span (example keys: `session`, `env`, `system`, `version`, and so on). For more information about tags, see [Getting Started with Tags][9].

Reviewer 2: ```suggestion
<br />A dictionary of JSON serializable key-value pairs that users can add as tags on the span. Example keys: `session`, `env`, `system`, and `version`. For more information about tags, see [Getting Started with Tags][9].
```

> <br />A dictionary of JSON serializable key-value pairs that users can add as tags regarding the span's context (`session`, `environment`, `system`, `versioning`, and so on). For more information about tags, see [Getting Started with Tags][9].

Reviewer: ```suggestion
<br />A dictionary of JSON serializable key-value pairs that users can add as tags on the span (example: `session`, `env`, `system`, `version`, and so on). For more information about tags, see [Getting Started with Tags][9].
```

### Example

{{< code-block lang="python" >}}
from ddtrace.llmobs import LLMObs
from ddtrace.llmobs.decorators import workflow

> @workflow(model="model_name", model_provider="model_provider")

Reviewer: ```suggestion
@workflow
```
Workflow spans don't take model name/provider AFAIK
